core/trino-main/src/main/java/io/trino/operator/output/RowPositionsAppender.java

```diff
@@ -66,13 +66,14 @@ private RowPositionsAppender(PositionsAppender[] fieldAppenders, int expectedPos
     }
 
     @Override
+    // TODO: Make PositionsAppender work performant with different block types (https://github.com/trinodb/trino/issues/13267)
     public void append(IntArrayList positions, Block block)
     {
         if (positions.isEmpty()) {
             return;
         }
         ensureCapacity(positions.size());
-        RowBlock sourceRowBlock = (RowBlock) block;
+        AbstractRowBlock sourceRowBlock = (AbstractRowBlock) block;
```

Comment:
There are no other implementations of AbstractRowBlock aside from RowBlockBuilder, which would never appear here.

Reply:
There could be user implementations... It's SPI